src/NServiceBus.Core.Tests/Correlation/CorrelationContextExtensionsTests.cs

```diff
@@ -0,0 +1,52 @@
+﻿namespace NServiceBus.Core.Tests.Correlation
+{
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class CorrelationContextExtensionsTests
+    {
+        [Test]
+        public void SendOptions_GetCorrelationId_Should_Return_Configured_CorrelationId()
```

Comment:
While reading this I realized how many test naming obscurities we have. I.ex. I'd written only the names of the framework in the real casing like SendOptions and the rest lower case. Doesn't matter for this PR and I think we should have some kind of guidelined for this for future PRs.